File: app/src/main/java/com/lugo/manueln/socialapp/Presenters/PresenterPostComplete.java
package com.lugo.manueln.socialapp.Presenters;

import android.support.v4.app.FragmentActivity;

import com.lugo.manueln.socialapp.Interactors.ModulePostComplete;
import com.lugo.manueln.socialapp.Interfaces.interPostComplete;
import com.lugo.manueln.socialapp.objects.Comments;
import com.lugo.manueln.socialapp.objects.Post;

import java.util.List;

public class PresenterPostComplete implements interPostComplete.presenter{

    interPostComplete.view view;
    interPostComplete.interactor interactor;
    public PresenterPostComplete(interPostComplete.view view) {

        this.view=view;
        interactor=new ModulePostComplete(this);
    }

    @Override
    public void loadPostCompleteWithComments(int id, FragmentActivity fragmentActivity) {

        interactor.retrofitGetPostCompleteWithComments(id,fragmentActivity);
    }

    @Override
    public void PostCompleteWithCommentsPresenter(Post miPost, List<Comments> commentsList) {

        if(view!=null){
            view.showPostComplete(miPost);
            view.showCommentsPost(commentsList);
        }
    }

    @Override
    public void sendErrorLoadPost(String error) {

        if(view!=null){

            view.showErrorLoadPost(error);
        }

    }

    @Override
    public void saveComment(Comments comment) {

        interactor.saveCommentPost(comment);

    }

    @Override
    public void updatePostComplete() {


    }
}

File: app/src/main/java/com/lugo/manueln/socialapp/presentation/PostComplete/Presenter/PresenterPostComplete.kt
package com.lugo.manueln.socialapp.presentation.PostComplete.Presenter

import android.support.v4.app.FragmentActivity

import com.lugo.manueln.socialapp.domain.interactors.postCompleteInteractor.InteractorPostCompleteImpl
import com.lugo.manueln.socialapp.domain.interactors.postCompleteInteractor.InteractorPostComplete
import com.lugo.manueln.socialapp.domain.Comments
import com.lugo.manueln.socialapp.domain.Post
import com.lugo.manueln.socialapp.presentation.PostComplete.PostCompleteContract
import com.lugo.manueln.socialapp.usecases.PostComplete.GetCommentsPost
import com.lugo.manueln.socialapp.usecases.PostComplete.GetPostComplete
import io.reactivex.Observer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.disposables.Disposable
import io.reactivex.schedulers.Schedulers

class PresenterPostComplete(private var view: PostCompleteContract.view?,private val getPostComplete: GetPostComplete,private val getCommentsPost:GetCommentsPost) : PostCompleteContract.presenter {


    override fun loadPostCompleteWithComments(id: Int) {

        view?.showProgressBar()

        getPostComplete.invoke(id).subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object :Observer<Post>{
            override fun onComplete() {

            }

            override fun onSubscribe(d: Disposable) {

            }

            override fun onNext(post: Post) {

                if(view!=null){
                    view?.showPostComplete(post)
                }


            }

            override fun onError(e: Throwable) {

                if(view!=null){
                    view?.showErrorLoadPost(e.message)
                }
            }
        })


        getCommentsPost.invoke(id).subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(object :Observer<List<Comments>>{
                    override fun onComplete() {

                    }

                    override fun onSubscribe(d: Disposable) {

                    }

                    override fun onNext(listComments: List<Comments>) {

                        if(view!=null){
                            view?.showCommentsPost(listComments)

                            view?.hideProgressbar()
                        }

                    }

                    override fun onError(e: Throwable) {

                        if(view!=null){
                         view?.showErrorLoadPost(e.message)
                        }
                    }
                })



    }

    override fun PostCompleteWithCommentsPresenter(miPost: Post, commentsList: List<Comments>) {

        if (view != null) {
            view!!.showPostComplete(miPost)
            view!!.showCommentsPost(commentsList)
        }
    }

    override fun sendErrorLoadPost(error: String?) {

        if (view != null) {

            view!!.showErrorLoadPost(error)
        }

    }

    override fun saveComment(comment: Comments) {



    }

    override fun updatePostComplete() {

    }
}
